lib/modules/apostrophe-images/index.js

```diff
@@ -80,8 +80,73 @@ module.exports = {
       }
     ].concat(options.arrangeFields || []);
 
+    options.addFilters = [
+      {
+        name: 'orientation',
+        choices: [
+          {
+            value: 'landscape',
+            label: 'Lansdscape'
+          },
+          {
+            value: 'portrait',
+            label: 'Portrait'
+          },
+          {
+            value: 'square',
+            label: 'Square'
+          }
+        ],
+        def: null
+      }
+    ].concat(options.addFilters || []);
+
   },
   construct: function(self, options) {
+
+    const fileImages = self.apos.attachments.fileGroups
+      .find((group) => group.name === 'images');
+
+    options.addFilters = [
+      ...options.addFilters || [],
+      {
+        name: 'fileType',
+        choices: fileImages.extensions.map((ext) => ({
+          value: ext,
+          label: ext.charAt(0).toUpperCase() + ext.slice(1)
+        })),
+        def: null
+      }
+    ];
+
+    self.on('apostrophe:modulesReady', 'getMediaSourcesModules', () => {
```

Comment:
This filter does not belong in the core. It should be added via the `improve` mechanism in the media sources bundle. The others are fine in the core because they do not require an optional module.

Reply:
Oh ok, yes you right, I remove this PR so.